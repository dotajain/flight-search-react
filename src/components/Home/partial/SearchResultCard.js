import React from 'react';
import PropTypes from 'prop-types';

import FlightSearchInfo from './FlightSearchInfo';

const SearchResultCard = ({ result }) => {
  let totalPrice = result.onWardFlight.fare;

  let renderOnwardCard;
  let renderReturnCard;

  if (result.onWardFlight) {
    renderOnwardCard = <FlightSearchInfo result={result.onWardFlight} />;
  }

  if (result.returnFlight) {
    totalPrice += result.returnFlight.fare;
    renderReturnCard = <FlightSearchInfo result={result.returnFlight} />;
  }

  const toCurrency = number =>
    Number(number)
      .toFixed(0)
      .replace(/(\d)(?=(\d{3}))/g, '$1,');

  return (
    <div className="box box__card">
      <div className="row">
        <div className="col">
          <div className="price"> Rs. {toCurrency(totalPrice)} </div>

          <div className="row">
            {renderOnwardCard}
            {renderReturnCard}
          </div>
        </div>
        <div className="col search__result--flight-logo">
          <div className="box">
            <div className="flight__logo">
              <div className="flight__logo--box">Flight Logo</div>
            </div>
            <button className="flight--btn">Book this flight</button>
          </div>
        </div>
      </div>
    </div>
  );
};

SearchResultCard.propTypes = {
  result: PropTypes.object,
};

export default SearchResultCard;
