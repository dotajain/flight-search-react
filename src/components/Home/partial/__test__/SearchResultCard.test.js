import React from 'react';
import renderer from 'react-test-renderer';

import SearchResultCard from '../SearchResultCard';

global.fetch = jest.fn(
  () =>
    new Promise(resolve => {
      process.nextTick(() => resolve({ json: () => ({}) }));
    }),
);

it('Properly rendered SearchResultCard Component', () => {
  const component = renderer.create(<SearchResultCard result={result}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


const result = {
  'onWardFlight': {
    'airline': {
      'name': 'Vistara',
      'code': '993',
      'number': 'UK',
    },
    'origin': {
      'airportCode': 'DEL',
      'terminal': '3',
      'time': '12:50',
    },
    'destination': {
      'airportCode': 'BOM',
      'time': '14:55',
    },
    'bookingclass': 'E',
    'depdate': '2017-09-15t1250',
    'arrdate': '2017-09-15t1455',
    'travelTime': '2h 5m',
    'fare': 2986,
  },
  'returnFlight': {
    'airline': {
      'name': 'Vistara',
      'code': '993',
      'number': 'UK',
    },
    'origin': {
      'airportCode': 'DEL',
      'terminal': '3',
      'time': '12:50',
    },
    'destination': {
      'airportCode': 'BOM',
      'time': '14:55',
    },
    'bookingclass': 'E',
    'depdate': '2017-09-15t1250',
    'arrdate': '2017-09-15t1455',
    'travelTime': '2h 5m',
    'fare': 2986,
  },
};