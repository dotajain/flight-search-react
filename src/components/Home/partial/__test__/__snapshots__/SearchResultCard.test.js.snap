// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly rendered SearchResultCard Component 1`] = `
<div
  className="box box__card"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="price"
      >
         Rs. 
        5,972
         
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="flight"
          >
            <small
              className="flight__number"
            >
              993 - UK
            </small>
            <div
              className="flight__airport-code"
            >
              DEL &gt; BOM
            </div>
            <div
              className="flight__time"
            >
              <span
                className="flight__time--label"
              >
                Depart:
              </span>
              <span
                className="flight__time--value"
              >
                12:50
              </span>
            </div>
            <div
              className="flight__time"
            >
              <span
                className="flight__time--label"
              >
                Arrive:
              </span>
              <span
                className="flight__time--value"
              >
                14:55
              </span>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="flight"
          >
            <small
              className="flight__number"
            >
              993 - UK
            </small>
            <div
              className="flight__airport-code"
            >
              DEL &gt; BOM
            </div>
            <div
              className="flight__time"
            >
              <span
                className="flight__time--label"
              >
                Depart:
              </span>
              <span
                className="flight__time--value"
              >
                12:50
              </span>
            </div>
            <div
              className="flight__time"
            >
              <span
                className="flight__time--label"
              >
                Arrive:
              </span>
              <span
                className="flight__time--value"
              >
                14:55
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col search__result--flight-logo"
    >
      <div
        className="box"
      >
        <div
          className="flight__logo"
        >
          <div
            className="flight__logo--box"
          >
            Flight Logo
          </div>
        </div>
        <button
          className="flight--btn"
        >
          Book this flight
        </button>
      </div>
    </div>
  </div>
</div>
`;
